From 9fed03e5d9ae0df7572d585c3538d2c24216adb7 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 12 Nov 2020 19:11:43 +0200
Subject: [PATCH] [SW-24179] preboot: goya2: unset const for arc images copy

If Scratchpad SRAM is read with out writing to it, causes ECC.
In legacy+ boot flow, memcopy block is used to copy ARC ICCM and DCCM
images to SP SRAM as part of boot fit load.
Un-aligned address writes are causing ECC.

This patch set const value of memcopy block to zero explicitly to
make sure actual ICCM and DCCM images are copied instead of all zeros.

Change-Id: Ib01f24a73f6b41642025fe46d4fd39817f6ae6c1
---

diff --git a/goya2/common.c b/goya2/common.c
index 8691254..52c43d0 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -426,6 +426,7 @@
 	/* Set length in memcopy register in granuality of 16 bytes */
 	len = len >> 4;
 	mem_cfg = regmap_readl(mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
+	mem_cfg &= ~(PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_MASK);
 	mem_cfg &= ~(PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK);
 	mem_cfg |= (len & PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK);
 	regmap_writel(mem_cfg, mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
